RELEASE_GUIDE.md

```diff
@@ -42,50 +70,134 @@ considering what effects this may have:
 - JSON produced by `purs publish`
   - this might affect Pursuit
 
+Collaborate with other maintainers to identify what work needs to be done to resolve any issue.
+
 ## Making a release
 
-- Make a commit bumping versions. The following should be updated:
+To make a release, a maintainer should create a PR updating versions in the repo, and once that is merged, they can create a GitHub release, which triggers a CI build that should upload prebuilt binaries into that GitHub release page. (If it doesn't, you'll upload those manually).
+
+After that, the maintainer will upload the respective packages to Hackage and NPM.
+
+### Make last changes and submit the PR
+
+1. Checkout a new branch based on current `master`
+
+```bash
+git fetch upstream
+git checkout upstream/master
+git switch -c prepareNextRelease
+```
+
+2. Regenerate the LICENSE file and commit it. See `./license-generator/` for details.
+
+```bash
+make license-generator
+git add LICENSE
+git commit -m "Regenerated LICENSE file"
+```
+
+3. Edit the CHANGELOG.md file
+    - Create a new "Unreleased" section
+    - Rename the old "Unreleased" section to the version you will be releasing
+    - Edit the new version's content until satisfied (e.g. grammar, spelling, clarity, etc.)
+
+```bash
+git add CHANGELOG.md
+git commit -m "Finalizing CHANGELOG for {version}"
+```
+
+4. Make a commit bumping versions. The following should be updated:
+
+- If `purescript-cst` has changed at all since the last release:
+
+    - In `lib/purescript-cst/purescript-cst.cabal`, update the `version` field
+      (note that the new version should be based on the PVP, according to what
+      changed since the previous release, and not on the actual compiler
+      version)
+
+    - In `purescript.cabal`, update the version bounds for `purescript-cst`
 
-  - The `version` field in `purescript.cabal`
+  - In `lib/purescript-cst/README.md`, update the versions table to include a new entry for the new PureScript release
 
-  - The `prerelease` field in `app/Version.hs`, if updating the prerelease
-    field
+  - In `purescript.cabal`, update the `version` field
```

Comment:
This item and all of the other items following it need to be dedented so they aren't sub-items of the "If purescript-cst has changed" part, because all of these things need to happen every time, regardless of whether purescript-cst has changed.